### PR TITLE
Add more EEAs to Java collection objects

### PR DESCRIPTION
**Pull request by wborn**
While using the EEAs I was missing annotations on some commonly used collection classes.

So I've added more annotations in this PR for:

* ArrayList
* Collection
* HashMap
* HashSet
* LinkedList
* List
* Map
* Set
* ConcurrentHashMap

### CODE REVIEW

**Comment by jason-faust** [comment]
@J-N-K Eyeballing this it looks ok, how do you want to handle accepting this?

File: libraries/java/java/util/Set.eea [hunk 1]
```diff
@@ -1,7 +1,67 @@
...
+ ()Ljava/util/Spliterator<TE;>;
+ ()L1java/util/Spliterator<TE;>;
 toArray
  ()[Ljava/lang/Object;
- ()[1Ljava/lang/Object;
+ ()[1L1java/lang/Object;
 toArray
  <T:Ljava/lang/Object;>([TT;)[TT;
- <T:Ljava/lang/Object;>([TT;)[1TT;
+ <T:Ljava/lang/Object;>([T1T;)[1T1T;
```
Comment (J-N-K) [inline libraries/java/java/util/Set.eea:67]: See above. Depends on the implementation. In general null-elements in sets are allowed, so a restriction here seems to be wrong.

**Review by J-N-K — CHANGES_REQUESTED** [review]

**Comment by J-N-K** [comment]
Thanks @wborn

File: libraries/java/java/util/ArrayList.eea [hunk 2]
```diff
@@ -0,0 +1,91 @@
+class java/util/ArrayList
+<init>
+ (Ljava/util/Collection<+TE;>;)V
+ (L1java/util/Collection<+TE;>;)V
+add
+ (TE;[Ljava/lang/Object;I)V
+ (TE;[L1java/lang/Object;I)V
```
Comment (J-N-K) [inline libraries/java/java/util/ArrayList.eea:7]: Isn't that method private?

File: libraries/java/java/util/Collection.eea [hunk 3]
```diff
@@ -1,13 +1,40 @@
...
+spliterator
+ ()Ljava/util/Spliterator<TE;>;
+ ()L1java/util/Spliterator<TE;>;
 stream
  ()Ljava/util/stream/Stream<TE;>;
  ()L1java/util/stream/Stream<TE;>;
 toArray
  ()[Ljava/lang/Object;
- ()[1Ljava/lang/Object;
+ ()[1L1java/lang/Object;
```
Comment (J-N-K) [inline libraries/java/java/util/Collection.eea:34]: `Collection` in general allows `null` elements, so a non-null-array of (infered nullable) `Object` seems to be ok

File: libraries/java/java/util/Collection.eea [hunk 4]
```diff
@@ -1,13 +1,40 @@
...
+toArray
+ <T:Ljava/lang/Object;>(Ljava/util/function/IntFunction<[TT;>;)[TT;
+ <T:Ljava/lang/Object;>(L1java/util/function/IntFunction<[TT;>;)[T1T;
```
Comment (J-N-K) [inline libraries/java/java/util/Collection.eea:37]:
```suggestion
 <T:Ljava/lang/Object;>(L1java/util/function/IntFunction<[1TT;>;)[1TT;
```

The generator needs to return a non-null-array of the desired type, so if the type is @Nullable, members of the array can be null, the array itself is non-null.

File: libraries/java/java/util/LinkedList.eea [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ (L0java/lang/Object;)Z
+spliterator
+ ()Ljava/util/Spliterator<TE;>;
+ ()L1java/util/Spliterator<TE;>;
+toArray
+ ()[Ljava/lang/Object;
+ ()[L1java/lang/Object;
+toArray
+ <T:Ljava/lang/Object;>([TT;)[TT;
+ <T:Ljava/lang/Object;>([T1T;)[T1T;
```
Comment (J-N-K) [inline libraries/java/java/util/LinkedList.eea:70]: see above

**Comment by wborn** [comment]
I think adding annotations to arrays in Eclipse is a bit buggy.

**Review by J-N-K — APPROVED** [review]

**Pull request merged**